### PR TITLE
TT-7899 API span attributes

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Adding span attributes to main GW span + VersionCheckMW attributes +  AuthKey MW attributes + OauthMW attributes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit:  
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578024128)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: f0cdd4f9f27c8c1e3373542f0535d95ea286fcdd 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578167464)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: f8e3b3e5e3f04bcbf856bddc8256b90d08d0f671 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578453545)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: 6b536008d24bb43e8f2bc631cf21dc4862e0616f 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578458150)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: c3ee32f0bc349a7427088658e79078f7854e5f81 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578485241)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: e7ccfc9bc4717781d80ae0ebb35f31783ef10ec5 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5578576881)

File: internal/otel/otel_test.go [hunk 1]
```diff
@@ -101,3 +103,117 @@ func Test_InitOpenTelemetry(t *testing.T) {
...
+		version = "Non Versioned"
+	}
+	return semconv.TykAPIVersion(version)
+}
+
+var APIKeyAttribute = semconv.TykAPIKey
+
+var OAuthClientIDAttribute = semconv.TykOauthID
+
+*/
```
Comment (sredxny) [inline internal/otel/otel_test.go:137]: remove?

File: gateway/mw_auth_key.go [hunk 2]
```diff
@@ -145,6 +146,7 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 	case apidef.AuthToken, apidef.UnsetAuth:
 		ctxSetSession(r, &session, updateSession, k.Gw.GetConfig().HashKeys)
 		k.setContextVars(r, key)
+		ctxSetSpanAttributes(r, k.Name(), otel.APIKeyAttribute(key))
```
Comment (sredxny) [inline gateway/mw_auth_key.go:149]: are we exposing here the key? I think we should avoid the key to be displayed in any system unless hashed or blurred

File: internal/otel/otel.go [hunk 3]
```diff
@@ -48,3 +59,33 @@ func InitOpenTelemetry(ctx context.Context, logger *logrus.Logger, gwConfig *Con
...
+
+		attrs = append(attrs, semconv.TykAPITags(tags...))
+	}
+
+	return attrs
+}
+
+func APIVersionAttribute(version string) SpanAttribute {
+	if version == "" {
+		version = "Non Versioned"
```
Comment (sredxny) [inline internal/otel/otel.go:84]: what about moving this string ("Non Versioned") into a constant?

**Review by sredxny — CHANGES_REQUESTED** [review]

File: gateway/mw_auth_key.go (hunk 2 above)
Comment (tbuchaillot) [reply gateway/mw_auth_key.go:149]: On Tyk Pump we are sending the raw key - I'll ask around. Good question!

File: internal/otel/otel.go (hunk 3 above)
Comment (tbuchaillot) [reply internal/otel/otel.go:84]: Done!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: f5ff5a6904456d299efde1543dccc4e25695ee5d 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5585806516)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5331/merge
Commit: 9027b698cca44a1b410c3be26088b0fecfdabbc2 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5586699871)

**Review by sredxny — APPROVED** [review]

**Pull request merged**